app/views/instrument_price_policies/_price_policy_fields.html.haml

```diff
@@ -5,7 +5,7 @@
 
 - price_policy = @price_policies.first
 = f.input :charge_for,
-  collection: charge_for_options(price_policy.product),
+  collection: local_assigns[:collection] ? collection : charge_for_options(price_policy.product),
```

Comment:
I would lean towards calling the option something like `charge_for_collection`, and if it's not there, don't show the dropdown at all.